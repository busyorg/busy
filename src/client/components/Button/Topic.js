import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { Link } from 'react-router-dom';
import './Topic.less';

class Topic extends React.Component {
  static propTypes = {
    name: PropTypes.string,
    favorite: PropTypes.bool,
    closable: PropTypes.bool,
    onClose: PropTypes.func,
  };

  static defaultProps = {
    name: '',
    favorite: false,
    closable: false,
    onClose: () => {},
  };

  state = {
    closing: false,
  };

  onCloseOver = () => {
    this.setState({
      closing: true,
    });
  };

  onCloseOut = () => {
    this.setState({
      closing: false,
    });
  };

  onCloseClick = e => {
    e.preventDefault();
    this.props.onClose(this.props.name);
  };

  render() {
    const { name, favorite, closable } = this.props;

    return (
      <Link
        className={classNames('Topic', {
          'Topic--favorite': favorite,
          'Topic--closing': this.state.closing,
        })}
        to={`/trending/${name}`}
      >
        {name}
        {closable && (
          <i
            role="presentation"
            className="iconfont icon-close Topic__close"
            onClick={this.onCloseClick}
            onMouseOver={this.onCloseOver}
            onMouseOut={this.onCloseOut}
          />
        )}
      </Link>
    );
  }
}

export default Topic;
