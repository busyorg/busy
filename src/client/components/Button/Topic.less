@import (reference) "../../styles/custom.less";
@import (reference) "../../styles/modules/variables.less";
@import (reference) "../../styles/mixins.less";

.Topic {
  .btn;
  .btn-primary;
  .btn-ghost;

  .colorize-ghost-button(@green);

  height: @btn-height-sm;
  font-size: 14px;
  border-radius: @btn-height-sm;
  border-width: 2px;
  padding: 0 8px;
  line-height: 21px;

  text-decoration: none !important;

  @media @small {
    padding: 0 10px;
  }

  & > &__close {
    font-size: 14px !important;
    vertical-align: -2px;
    margin-left: 4px;
  }
}

.Topic--favorite {
  .colorize-ghost-button(@orange)
}

.Topic--closing {
  .colorize-ghost-button(@red);
}
