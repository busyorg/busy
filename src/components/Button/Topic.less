@import "../../styles/custom.less";
@import "../../styles/mixins.less";

.Topic {
  .btn;
  .btn-primary;
  .btn-ghost;

  .colorize-ghost-button(@green);

  height: @btn-height-sm;
  font-size: 14px;
  border-radius: @btn-height-sm;
  border-width: 2px;
  padding: 0 10px;
  line-height: 21px;

  text-decoration: none !important;

  & > &__close {
    font-size: 14px !important;
    vertical-align: -2px;
    margin-left: 4px;
  }
}

.Topic--favorite {
  .colorize-ghost-button(@orange)
}

.Topic--closing {
  .colorize-ghost-button(@red);
}
