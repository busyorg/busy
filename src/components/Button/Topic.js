import React, { PropTypes } from 'react';
import { Link } from 'react-router';
import './Topic.less';

const Topic = ({ name, isFavorite }) =>
  <Link to={`/trending/${name}`} className={`Topic ${(isFavorite) ? 'Topic--favorite' : ''}`}>
    {name}
  </Link>;

Topic.propTypes = {
  name: PropTypes.string,
  isFavorite: PropTypes.bool
};

export default Topic;
