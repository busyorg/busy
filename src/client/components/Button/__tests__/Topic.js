import React from 'react';
import { shallow } from 'enzyme';
import Topic from '../Topic';

describe('<Topic />', () => {
  it('renders without exploding', () => {
    const props = {
      name: 'exampletopic',
      favorite: true,
      closable: true,
    };
    const wrapper = shallow(<Topic {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
