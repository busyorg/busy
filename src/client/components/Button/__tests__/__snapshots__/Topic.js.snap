// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Topic /> renders without exploding 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Topic
    closable={true}
    favorite={true}
    name="exampletopic"
    onClose={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        "exampletopic",
        <i
          className="iconfont icon-close Topic__close"
          onClick={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          role="presentation"
        />,
      ],
      "className": "Topic Topic--favorite",
      "replace": false,
      "to": "/trending/exampletopic",
    },
    "ref": null,
    "rendered": Array [
      "exampletopic",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "iconfont icon-close Topic__close",
          "onClick": [Function],
          "onMouseOut": [Function],
          "onMouseOver": [Function],
          "role": "presentation",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          "exampletopic",
          <i
            className="iconfont icon-close Topic__close"
            onClick={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
            role="presentation"
          />,
        ],
        "className": "Topic Topic--favorite",
        "replace": false,
        "to": "/trending/exampletopic",
      },
      "ref": null,
      "rendered": Array [
        "exampletopic",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "iconfont icon-close Topic__close",
            "onClick": [Function],
            "onMouseOut": [Function],
            "onMouseOver": [Function],
            "role": "presentation",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
